features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/vitals/JankStatsActivityLifecycleListener.kt

```diff
@@ -160,6 +195,42 @@ internal class JankStatsActivityLifecycleListener(
         activeActivities[window] = list
     }
 
+    @RequiresApi(Build.VERSION_CODES.N)
+    private fun registerMetricListener(window: Window, activity: Activity) {
+        if (buildSdkVersionProvider.version() >= Build.VERSION_CODES.S) {
+            if (frameMetricsListener == null) {
+                frameMetricsListener = DDFrameMetricsListener()
+            }
+            val handler = Handler(Looper.getMainLooper())
+            frameMetricsListener?.let { window.addOnFrameMetricsAvailableListener(it, handler) }
+        } else {
+        // Fallback - Android 30 allows apps to not run at a fixed 60hz, but didn't yet have
+        // Frame Metrics callbacks available
+        if (display == null && buildSdkVersionProvider.version() == Build.VERSION_CODES.R) {
+            val displayManager = activity.getSystemService(Context.DISPLAY_SERVICE) as DisplayManager
+            display = displayManager.getDisplay(Display.DEFAULT_DISPLAY)
+        }
```

Comment:
maybe we can pull this out from this method? it may be confusing, because there is no listener registration in this block. Also by pulling it out we will have a symmetry at the calls site between `registerMetricListener` and `unregisterMetricListener` calls in terms of the build version required, because currently the former is guarded by `Build.VERSION.SDK_INT >= Build.VERSION_CODES.R` and the latter `Build.VERSION.SDK_INT >= Build.VERSION_CODES.S` - this raises the obvious question initially: we don't unregister listener registered if `Build.VERSION.SDK_INT == Build.VERSION_CODES.R`?

After this change, the following:

```
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            registerMetricListener(window, activity)
        }
```

will become:

```
        if (buildSdkVersionProvider.version() >= Build.VERSION_CODES.S) {
            registerMetricListener(window, activity)
        } else if (display == null && buildSdkVersionProvider.version() == Build.VERSION_CODES.R) {
            val displayManager = activity.getSystemService(Context.DISPLAY_SERVICE) as DisplayManager
            display = displayManager.getDisplay(Display.DEFAULT_DISPLAY)
        }
```

which is cleaner to me